src/model/deck-widget-models.ts

```diff
@@ -0,0 +1,327 @@
+import type { WidgetModel } from "@jupyter-widgets/base";
+import { BaseModel } from "./base.js";
+import { Widget, WidgetPlacement } from "@deck.gl/core";
+import { CompassWidget, ZoomWidget, FullscreenWidget, LightTheme } from "@deck.gl/widgets";
+import { LegendWidget, NorthArrowWidget, TitleWidget, ScaleWidget, SaveImageWidget } from "./deck-widget.js";
+
+export abstract class BaseDeckWidgetModel extends BaseModel {
```

Comment:
At first glance, I wouldn't think these should be models... do the widgets themselves need to be reactive on the Jupyter side?